### PR TITLE
Resolves staticcheck failures for component-base/metrics

### PR DESCRIPTION
**Pull request by maryamklabib**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Resolves staticcheck failures for component-base/metrics:
```
Errors from staticcheck:
vendor/k8s.io/component-base/metrics/metric.go:193:5: var noopSummaryVec is unused (U1000)
vendor/k8s.io/component-base/metrics/processstarttime.go:57:18: p.NewStat is deprecated: use p.Stat() instead  (SA1019)
```
[We have verified that the unused variable is not used in any other components](https://cs.k8s.io/?q=noopSummaryVec&i=nope&files=&repos=).

We replaced the deprecated method with the recommended one.

**Which issue(s) this PR fixes**:
https://github.com/kubernetes/kubernetes/issues/92402

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

cc @nikhita

### CODE REVIEW

**Comment by maryamklabib** [comment]
I signed the CLA.

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by nikhita** [comment]
/ok-to-test
/priority backlog

/assign @dims
for hack/ approval

**Comment by nikhita** [comment]
@maryamklabib Congrats on getting your first PR out to @kubernetes! :tada: 

Just a quick note that even after this PR gets approved, it won't get merged right away since we are in code freeze for the 1.19 release. Once the code freeze is lifted, it'll get merged automatically.

**Comment by dims** [comment]
/retest
/approve
/lgtm

thanks @maryamklabib !

**Comment by dims** [comment]
Looks like this will merge when master opens for business

**Pull request merged**
